pkg/controller/elasticsearch/driver/upgrade_predicates.go

```diff
@@ -27,7 +29,33 @@ type Predicate struct {
 	fn   func(context PredicateContext, candidate corev1.Pod, deletedPods []corev1.Pod, maxUnavailableReached bool) (bool, error)
 }
 
+type failedPredicate struct {
+	pod       string
+	predicate string
+}
+
+func newFailedPredicate(pod, predicate string) *failedPredicate {
```

Comment:
Not sure if the non-exported constructor on a non-exported trivial struct adds much?

Reply:
Sorry for the drive-by comments!